README.md

```diff
@@ -117,11 +117,14 @@ Periodically development code snapshots are provided at the following URL:
 
 Source packages for current and previous releases are located at:
 
-   hdf5 1.14 releases:
-   https://support.hdfgroup.org/releases/hdf5/v1_14/index.html
+   HDF5 2.0.0 release:
```

Comment:
I really want to avoid having to update all these links with every new release. I think it is good enough to reference the latest release to the latest link.